tasks/main.yml

```diff
@@ -63,14 +63,61 @@
     state: "{{ item.state | default(omit) }}"
     __report_changed: "{{ not __firewall_previous_replaced }}"
   loop: "{{ firewall is mapping | ternary([firewall], firewall) |
-    map('dict2items') | map('difference', __previous) | select |
+    map('dict2items') | map('difference', __previous) |
+    map('difference', __detailed) | select |
     map('items2dict') | list }}"
   vars:
     __previous:
       - key: previous
         value: replaced
+    __detailed:
+      - key: detailed
+        value: true
+      - key: detailed
+        value: false
   register: firewall_lib_result
 
+- name: gather firewall ansible fact with args
+  vars:
+    fw: "{{ firewall is mapping | ternary([firewall], firewall) }}"
+  block:
+    - name: gather firewall config information
+      vars:
+        __previous:
+          - key: previous
+            value: replaced
+      firewall_lib_facts:
+        detailed: "{{ item.detailed | default(False) }}"
+      loop: "{{ fw | map('dict2items') | map('difference', __previous) |
+        select | map('items2dict') | list }}"
+      register: __firewalld_facts
+
+    - name: update firewalld_config fact
+      vars:
+        firewalld_facts: "{{ __firewalld_facts.results[0] }}"
+      set_fact:
+        cacheable: yes
+        firewall_config: "{{ firewalld_facts.firewall_config }}"
```

Comment:
New parameter for setting cacheable? This `set_fact` argument allows for permanent storage of the fact, but it also causes the variable to be created twice in two different scopes. If used with the default plugin (builtin.memory) it doesn't really do all that much besides use up the extra RAM.

Reply:
Can't we just leave this up to the user to configure the caching outside of the role?  Do we have to indicate `cacheable` here?